src/cmd/serial.js

```diff
@@ -233,133 +246,265 @@ module.exports = class SerialCommand extends CLICommandBase {
+							macAddress = hwAddressByteArray;
+							currIfaceName = type;
+						}
+					}
+				}
+			}
 
-				return this.getDeviceMacAddress(device);
-			})
-			.then(data => {
-				console.log();
-				console.log('Your device MAC address is', chalk.bold.cyan(data));
-			})
-			.catch((err) => {
-				throw new VError(ensureError(err), 'Could not get MAC address');
-			});
+			// Print output
+			if (macAddress) {
+				this.ui.stdout.write(`Your device MAC address is ${chalk.bold.cyan(macAddress.slice(0, 6).map(num => num.toString(16).padStart(2, '0')).join(':'))}${os.EOL}`);
+				this.ui.stdout.write(`Interface is ${_.capitalize(currIfaceName)}${os.EOL}`);
+			} else {
+				this.ui.stdout.write(`Your device MAC address is ${chalk.bold.cyan('00:00:00:00:00:00')}${os.EOL}`);
+			}
+		} catch (err) {
+			throw new VError(ensureError(err), 'Could not get MAC address');
+		} finally {
+			if (device && device.isOpen) {
+				await device.close();
+			}
+		}
 	}
 
-	inspectDevice({ port }){
-		return this.whatSerialPortDidYouMean(port, true)
-			.then(device => {
-				if (!device){
-					throw new VError('No serial port identified');
-				}
+	/**
+	 * Inspects a Particle device and provides module info and asset info
+	 * @param {string} port
+	 */
+	async inspectDevice({ port }) {
+		let deviceFromSerialPort, deviceId, device;
+
+		try {
+			deviceFromSerialPort = await this.whatSerialPortDidYouMean(port, true);
+			deviceId = deviceFromSerialPort.deviceId;
+			device = await usbUtils.getOneUsbDevice({ idOrName: deviceId });
+		} catch (err) {
+			throw new VError(ensureError(err), 'Could not get inspect device');
+		}
 
-				return this.getSystemInformation(device);
-			})
-			.then((data) => {
-				const functionMap = {
-					s: 'System',
-					u: 'User',
-					b: 'Bootloader',
-					r: 'Reserved',
-					m: 'Monolithic',
-					a: 'Radio stack',
-					c: 'NCP'
-				};
-				const locationMap = {
-					m: 'main',
-					b: 'backup',
-					f: 'factory',
-					t: 'temp'
-				};
 
-				const d = JSON.parse(data);
-				const parser = new DescribeParser();
-				const modules = parser.getModules(d);
+		let platformName = null;
+		let platformId = null;
+		try {
+			platformName = deviceFromSerialPort.specs.productName;
+			platformId = deviceFromSerialPort.specs.productId;
+		} catch (err) {
+			// ignore error and move on to get other fields
+		}
+		this.ui.stdout.write(`Platform : ${platformId} - ${chalk.bold.cyan(platformName)}${os.EOL}${os.EOL}`);
+
+		try {
+			this._getModuleInfo(device);
+		} catch (err) {
+			throw new VError(ensureError(err), 'Could not get inspect device');
+		} finally {
+			if (device && device.isOpen) {
+				await device.close();
+			}
+		}
+	}
+	
+	/**
+	 * Obtains module info from control requests
+	 * @param {object} device
+	 * @returns {boolean} returns error or true
+	 */
+	async _getModuleInfo(device) {
+		// TODO: Get this from the protobuf exports
+		const fwModuleDisplayNames = {
```

Comment:
Idk if that's the meaning of your TODO (if so, just ignore this comment 😄 ) but there is in particle-usb an object that is exported with this data: https://github.com/particle-iot/particle-usb/blob/590d4e7caf43aedc8e14df3eeb1663d26d058905/src/device.js#L63C7-L63C33 you can use it instead